app/assets/javascripts/router.js

```diff
@@ -179,8 +187,20 @@ OSM.Router = function (map, rts) {
   };
 
   router.load = function () {
-    var loadState = currentRoute.run("load", currentPath);
+    var loadState = loadOrReload();
     router.stateChange(loadState || {});
+
+    function loadOrReload() {
+      if (currentRoute.has("reload") && window.performance) {
+        var wasReloaded = window.performance.getEntriesByType("navigation").some(function (entry) {
+          return entry.type === "reload";
+        });
```

Comment:
I'm not familiar with `window.performance` but from a quick bit of reading on MDN it seems this if testing if there are any navigation events of type `reload` in the history but what's the scope of that? The comment you added says we're interest in whether this request was a reload?

Reply:
I think it's for the current page only, not for the history.

https://developer.mozilla.org/en-US/docs/Web/API/Performance : "The Performance interface provides access to performance-related information for the current page".